chia/rpc/wallet_rpc_api.py

```diff
@@ -4496,3 +4504,83 @@ class CRCATApprovePending(Streamable):
         return {
             "transactions": [tx.to_json_dict_convenience(self.service.config) for tx in txs],
         }
+
+    ##########################################################################################
+    # VAULT
+    ##########################################################################################
+    @tx_endpoint(push=True)
+    async def vault_create(
+        self,
+        request: Dict[str, Any],
+        tx_config: TXConfig = DEFAULT_TX_CONFIG,
+        extra_conditions: Tuple[Condition, ...] = tuple(),
+    ) -> EndpointResult:
+        """
+        Create a new vault
+        """
+        assert self.service.wallet_state_manager
+        wallet = self.service.wallet_state_manager.main_wallet
+
+        secp_pk = bytes.fromhex(str(request.get("secp_pk")))
+        entropy = bytes.fromhex(str(request.get("entropy")))
```

Comment:
Can we call this "hidden_puzzle_hash" now?

Reply:
Actually, on further thought, let's take this out and come up with something canonical to use for the entropy of the hidden puzzle hash:
```
def calculate_hp_w_index_for_vault(index, hidden_puzzle = DEFAULT_HIDDEN_PUZZLE):
  return Program.to([6, (index, hidden_puzzle)])
```

Reply:
I think we may need to have some way to make the entropy secret but we can add that later.